metaflow/plugins/kfp/tests/README.md

```diff
@@ -0,0 +1,31 @@
+# Integration Tests for Metaflow on KFP
+
+These integration tests are based on Metaflow flows specifically designed the catch hard-to-find bugs. They can be run locally, or on Gitlab.
+
+# Local Tests
+
+To run the tests locally, first create your Metaflow config file locally, which can be found at: `/Users/[your_username]/.metaflowconfig/config.json`.
+This can also be obtained by running `metaflow configure show`. 
+
+A sample configuration:
+```
+{
+    "METAFLOW_DATASTORE_SYSROOT_S3": "s3://aip-example-dev/metaflow",
+    "METAFLOW_DEFAULT_DATASTORE": "local",
+    "KFP_RUN_URL_PREFIX": "https://kubeflow.corp.dev.zg-aip.net/",
+    "KFP_SDK_NAMESPACE": "aip-example",
+    "METAFLOW_USER": "hariharans@zillowgroup.com"
+}
+```
+
+Then, within the `tests` directory, run `python -m pytest -s -n 3 run_integration_tests.py --local`. The parameter `-n` specifies the number of parallel tests. You'll likely need to change this if resource constraints are an issue.
+
+# Github Tests
+
+These tests are configured to automatically run whenever you push a commit to the Github repository. A mirrored Gitlab repository detects changes, pulls in the changes, and triggers a pipeline.
+
+Currently, due to Gitlab's polling, it takes about 20 minutes for these tests to be triggered automatically. To trigger these tests manually, run:
+
+`curl -X POST "https://gitlab.zgtools.net/api/v4/projects/20508/mirror/pull?private_token=[PRIVATE_TOKEN]"`
```

Comment:
for ease of use, could you please create an executable bash script that is in the `tests` directory that runs this for us internally?

Reply:
@talebzeghmi what do you want this bash script to do?

Just run `curl -X POST ...`, or also pull the private token from Gitlab automatically?

Reply:
Declining to do this because it would expose the private Gitlab API token.